packages/expo/cli/utils/analytics/rudderstackClient.ts

```diff
@@ -0,0 +1,112 @@
+import RudderAnalytics from '@expo/rudder-sdk-node';
```

Comment:
@kgc00 can you review this part of the PR?

Reply:
Thanks for roping me in James, and good work on the PR Evan! Unfortunately, this would disrupt our current analytics in a few ways. Here are some problems I saw:
- we're using eas cli's write keys so events would end up in the wrong place
- we're now using a single client. Previously, expo-cli used two clients: a legacy XDL client and  a newer 'Unified' client. If we don't adopt that methodology events will also end up in the place.
- we're introducing a staging/prod distinction to events where there previously was only a prod pipeline. Probably a good idea in the long run but needs some backend setup and consideration if it's proper to do now.
- we're drops all tracking of events (we have the rudder client, but it is not used anywhere)

I also think it would be good to retain some common logic inside of a `Command` or `ExpoCommand` that all other commands subclass. This way we do not need to rely on developers remembering to add in track calls or initializing clients in the proper place when adding new commands.

I think it's a really good start, and I'm just thinking about how to address these issues. Will jump back in with suggestions later today when I have some answers.

Reply:
This is copied from eas-cli and reworked a bit to remove unused functions, structurally it's the same.

Reply:
hey! before i reply to some of the other comments:
> This is copied from eas-cli and reworked a bit to remove unused functions, structurally it's the same.

I have the feeling I am misunderstanding something, so sorry for the confusion. These analytics commands are expo-cli's commands, correct? e.g. this PR is not introducing eas-cli commands?

Reply:
@kgc00 both expo-cli and eas-cli implement identical commands for authentication, these copy the code from the newer eas-cli instead of the older expo-cli.

Reply:
> @kgc00 both expo-cli and eas-cli implement identical commands for authentication, these copy the code from the newer eas-cli instead of the older expo-cli.

cool, so based on your response I'm going to assume this PR is specifically for expo-cli events. In that case, I want to explain some differences between how analytics are set up between Expo CLI and EAS analytics:
- they use different write keys and end up in different places in our backends
  - Additionally, Expo CLI has two different clients which write to two different places in our backend
- Expo CLI has no distinction between staging and prod events

Just copying over how things are done in EAS will break existing pipelines (which I tried to point out in my first comments, sorry if that wasn't clear). I did speak with the folks using this data yesterday, so I will add in some suggestions shortly to fix these issues. We will be replicating the setup expo cli was using so we don't break event streams.

> I'd rather we import the analytics explicitly for every command we want to track rather than wrapping every command with a catch-all that could be (and based on expo-cli and eas-cli will be) bloated very quickly.

I think we might disagree here. The analytics in expo-cli were broken for who knows how long because they were initialized differently for each command. This led to trying to track events with no client available. I also think the eas entry point is quite clean, it does 2 things. The expo entry point is quite old, so I don't think it should be used as an example of best practices. I also think the additional time-to-start is worth the reduced complexity for having some location which can share global setup/state. But that's my two cents and I don't want to block you on this PR.

 > we don't track any events yet there is no start command in the CLI yet, please check usage
 
 Cool, thanks for clearing that up. Mind adding me in as a reviewer for future PRs touching analytics? thanks!

Reply:
spoke with Evan and some others outside of the pr, we'll be going with a lighter-weight approach (reflected in the updated suggestions).